src/lib/Depositor.sol

```diff
@@ -15,152 +14,48 @@ import {RewardsManagerCommon} from "./RewardsManagerCommon.sol";
-    );
-  }
-
-  /// @notice Preview the amount of undripped rewards that can be redeemed for `depositReceiptTokenAmount_` from a given
-  /// reward pool.
-  /// @param rewardPoolId_ The ID of the reward pool.
-  /// @param depositReceiptTokenAmount_ The amount of deposit receipt tokens to redeem.
-  /// @return rewardAssetAmount_ The amount of the reward pool's asset that can be redeemed.
-  function previewUndrippedRewardsRedemption(uint16 rewardPoolId_, uint256 depositReceiptTokenAmount_)
-    external
-    view
-    returns (uint256 rewardAssetAmount_)
-  {
+  function depositRewardAssetsWithoutTransfer(uint16 rewardPoolId_, uint256 rewardAssetAmount_) external {
     RewardPool storage rewardPool_ = rewardPools[rewardPoolId_];
-
-    rewardAssetAmount_ = _previewRedemption(
-      rewardPool_.depositReceiptToken,
-      depositReceiptTokenAmount_,
-      rewardPool_.dripModel,
-      rewardPool_.undrippedRewards,
-      rewardPool_.lastDripTime
-    );
+    _executeRewardDeposit(rewardPoolId_, rewardPool_.asset, rewardAssetAmount_, rewardPool_);
   }
 
   function _executeRewardDeposit(
     uint16 rewardPoolId_,
     IERC20 token_,
     uint256 rewardAssetAmount_,
-    address receiver_,
     RewardPool storage rewardPool_
-  ) internal returns (uint256 depositReceiptTokenAmount_) {
+  ) internal {
     if (rewardsManagerState == RewardsManagerState.PAUSED) revert InvalidState();
     _assertValidDepositBalance(token_, assetPools[token_].amount, rewardAssetAmount_);
 
-    IReceiptToken depositReceiptToken_ = rewardPool_.depositReceiptToken;
-
-    depositReceiptTokenAmount_ = RewardsManagerCalculationsLib.convertToReceiptTokenAmount(
-      rewardAssetAmount_, depositReceiptToken_.totalSupply(), _poolAmountWithFloor(rewardPool_.undrippedRewards)
-    );
-    if (depositReceiptTokenAmount_ == 0) revert RoundsToZero();
+    // To ensure reward drip times are in sync with reward deposit times we drip rewards before depositing.
+    _dripRewardPool(rewardPool_);
 
-    // Increment reward pool accounting only after calculating `depositReceiptTokenAmount_` to mint.
     rewardPool_.undrippedRewards += rewardAssetAmount_;
     assetPools[token_].amount += rewardAssetAmount_;
 
-    depositReceiptToken_.mint(receiver_, depositReceiptTokenAmount_);
-    emit Deposited(
-      msg.sender, receiver_, rewardPoolId_, depositReceiptToken_, rewardAssetAmount_, depositReceiptTokenAmount_
-    );
-  }
-
-  function _previewRedemption(
-    IReceiptToken receiptToken_,
-    uint256 receiptTokenAmount_,
-    IDripModel dripModel_,
-    uint256 totalPoolAmount_,
-    uint256 lastDripTime_
-  ) internal view returns (uint256 assetAmount_) {
-    uint256 nextDripAmount_ =
-      (lastDripTime_ != block.timestamp) ? _getNextDripAmount(totalPoolAmount_, dripModel_, lastDripTime_) : 0;
-    uint256 nextTotalPoolAmount_ = totalPoolAmount_ - nextDripAmount_;
```

Comment:
We need a `previewCurrentUndrippedRewards` for Dave that does this calculation and returns `nextTotalPoolAmount_`.

Reply:
[c573013](https://github.com/Cozy-Finance/cozy-safety-module-rewards-manager/pull/55/commits/c57301328cca804d4ea310d12659ee771b7a5803)